token/core/fabtoken/actions.go

```diff
@@ -122,26 +157,35 @@ func (t *TransferAction) GetOutputs() []driver.Output {
 	return res
 }
 
+// IsRedeemAt returns true if the output at the specified index is a redeemed output
 func (t *TransferAction) IsRedeemAt(index int) bool {
 	return t.Outputs[index].IsRedeem()
 }
 
+// IsGraphHiding returns false, indicating that fabtoken does not hide the transaction graph
 func (t *TransferAction) IsGraphHiding() bool {
 	return false
 }
 
+// SerializeOutputAt marshals the output at the specified index in TransferAction
 func (t *TransferAction) SerializeOutputAt(index int) ([]byte, error) {
+	if t.Outputs[index] == nil {
+		return nil, errors.New("failed to serialize output")
```

Comment:
add to this error that the output was nil

Reply:
done